application/apps/indexer/indexer_cli/src/main.rs

```diff
@@ -1799,3 +1799,60 @@ async fn detect_messages_type(input: &Path) -> Result<bool, DltParseError> {
         )))
     }
 }
+
+// Idea is to get the file path from the user,
+// read the first chunks from the file and return String
+// 'binary' or 'text' as of now. Later we can add enum for file
+// type.
+async fn detect_file_type(file_path: &Path) -> String {
+    let chunks = fetch_starting_chunk(file_path);
+    let buffer = match chunks {
+        Ok(buffer) => buffer,
+        // TODO: add correct error handling
+        Err(error) => panic!("Error while reading chunks from file: {}", error),
+    };
+    // println!("{:?}", buffer);
+    let result: bool = is_file_binary(buffer);
+
+    return String::from("Still in progress...");
+}
+
+fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>>{
+    let file = File::open(file_path)?;
+    let mut file = file.take(1024);
```

Comment:
In general, any static/configuration parameters keep on a top as const. It helps to easy manipulate with such kind of parameters 

```
const BYTES_TO_READ: usize = 1024;
...
let mut file = file.take(BYTES_TO_READ);
```

Reply:
better read in more then just 1k...better for the heuristics. maybe 10k